coyote.go

```diff
@@ -174,17 +178,32 @@ func main() {
 				log.Printf("💔 Terminating AMQP channel")
 			}()
 
+			persistent := ctx.Bool("persistent")
 			q, err := ch.QueueDeclare(
 				fmt.Sprintf("%s.%s", ctx.String("queue"), uuid.NewString()), // queue name
-				false, // is durable
-				true,  // is auto delete
-				true,  // is exclusive
-				false, // is no wait
-				nil,   // args
+				false,       // is durable
+				!persistent, // is auto delete
+				!persistent, // is exclusive
+				false,       // is no wait
+				nil,         // args
 			)
 			if err != nil {
 				return fmt.Errorf("%s %w", color.RedString("failed to declare a queue:"), err)
 			}
+			if persistent {
+				defer func() {
+					log.Printf("💔 Deleting queue %s", q.Name)
```

Comment:
This will delete the queue even if it is persistent right? Can you help me understand what is the benefit here?

Reply:
On a second look, I have an idea to use `--queue` argument to create persistent queue. Allow coyote to reconnect to an existing queue.


By default we generate interceptor queue names randomly with an UUID suffix:

```
interceptor.<uuid>
```

We can make it persistent if queue argument is provided.

```
coyote --queue <queue_name>
```

In this case coyote will try to bind to the queue if it exists or create it does not.

So persistent flag is not needed anymore. What do you think?

Reply:
You right! The logic here is coupled with a reconnection logic - if the queue is not persistent and we're disconnected from a cluster then RabbitMQ will automatically delete it.  
Do you think it might be useful to have an extra flag to enable/disable the queue auto deletion?

Reply:
For now I tied it to queue argument being present or not. Let's revisit with auto reconnect functionality.

Reply:
That's a brilliant idea, I like it! And indeed it makes sense to keep the queue even between restarts.